### ISSUE TITLE
amplify mock does not show graphql errors in response from lambda

### ISSUE THREAD

**Issue by aveeday** [aws-amplify/amplify-cli#3143]
**Describe the bug**
I use lambda resolver with ```@function(name: "function-${env}")```

In my function I validate data in request and throw error:
```js
throw new Error('validation error')
```

Then from appsync I can see error message in graphql response:
```json
{
  "errors": [
    {
      "message": "validation error",
      "errorType": "Lambda:Unhandled"
    }
  ]
}
```

But with amplify mock I do not receive error message:
```json
{
  "errors": [
    {
      "errorType": "null"
    }
  ]
}
```

**Amplify CLI Version**
4.6.0

**Comment by ammarkarachi** [comment]
@aveeday Thanks for bringing this up. The mock framework basically runs as a subprocess. We would need to bubble up the error from the process.

**Comment by hello2all** [comment]
It would be great if an example of error handling could be provided in the [transform docs](https://aws-amplify.github.io/docs/cli-toolchain/graphql#usage-1), and state error handling in mock is currently WIP.

**Comment by kaustavghosh06** [comment]
We released a fix for this with v4.14.1

**Comment by mgarabedian** [comment]
I am also having this issue. I am trying to figure out how I can throw a custom error with a code or type. In my lambda function I have the following:

`callback(new Error("Show Something"))`

My GQL result is this:

```
{
  "data": {
    "myFieldName": null
  },
  "errors": [
    {
      "message": {},
      "errorType": "Lambda:Unhandled",
      "data": null,
      "errorInfo": null,
      "path": [
        "myFieldName"
      ],
      "locations": [
        {
          "line": 3,
          "column": 3,
          "sourceName": "GraphQL request"
        }
      ]
    }
  ]
}
```

If I pass a string to 'callback' the message will show. But how can I throw a custom error with a code/type?
